interfaces/apparmor/apparmor_test.go

```diff
@@ -107,55 +107,49 @@ func (s *appArmorSuite) TestLoadProfilesRunsAppArmorParserReplaceWithSnapdDebug(
 
 // Tests for Profile.Unload()
 
-func (s *appArmorSuite) TestUnloadProfilesRunsAppArmorParserRemove(c *C) {
-	dirs.SetRootDir(c.MkDir())
-	defer dirs.SetRootDir("")
-	cmd := testutil.MockCommand(c, "apparmor_parser", "")
-	defer cmd.Restore()
-	err := apparmor.UnloadProfiles([]string{"snap.samba.smbd"}, dirs.AppArmorCacheDir)
-	c.Assert(err, IsNil)
-	c.Assert(cmd.Calls(), DeepEquals, [][]string{
-		{"apparmor_parser", "--remove", "snap.samba.smbd"},
-	})
-}
-
 func (s *appArmorSuite) TestUnloadProfilesMany(c *C) {
-	cmd := testutil.MockCommand(c, "apparmor_parser", "")
-	defer cmd.Restore()
 	err := apparmor.UnloadProfiles([]string{"/path/to/snap.samba.smbd", "/path/to/another.profile"}, dirs.AppArmorCacheDir)
 	c.Assert(err, IsNil)
-	c.Assert(cmd.Calls(), DeepEquals, [][]string{
-		{"apparmor_parser", "--remove", "/path/to/snap.samba.smbd", "/path/to/another.profile"},
-	})
 }
 
 func (s *appArmorSuite) TestUnloadProfilesNone(c *C) {
-	cmd := testutil.MockCommand(c, "apparmor_parser", "")
-	defer cmd.Restore()
 	err := apparmor.UnloadProfiles([]string{}, dirs.AppArmorCacheDir)
 	c.Assert(err, IsNil)
-	c.Check(cmd.Calls(), HasLen, 0)
 }
 
-func (s *appArmorSuite) TestUnloadProfilesReportsErrors(c *C) {
-	cmd := testutil.MockCommand(c, "apparmor_parser", "exit 42")
+func (s *appArmorSuite) TestUnloadRemovesCachedProfile(c *C) {
+	cmd := testutil.MockCommand(c, "apparmor_parser", "")
 	defer cmd.Restore()
-	err := apparmor.UnloadProfiles([]string{"snap.samba.smbd"}, dirs.AppArmorCacheDir)
-	c.Assert(err.Error(), Equals, `cannot unload apparmor profile: exit status 42
-apparmor_parser output:
-`)
+
+	dirs.SetRootDir(c.MkDir())
+	defer dirs.SetRootDir("")
+	err := os.MkdirAll(dirs.AppArmorCacheDir, 0755)
+	c.Assert(err, IsNil)
+
+	fname := filepath.Join(dirs.AppArmorCacheDir, "profile")
+	ioutil.WriteFile(fname, []byte("blob"), 0600)
+	err = apparmor.UnloadProfiles([]string{"profile"}, dirs.AppArmorCacheDir)
+	c.Assert(err, IsNil)
+	_, err = os.Stat(fname)
+	c.Check(os.IsNotExist(err), Equals, true)
 }
 
-func (s *appArmorSuite) TestUnloadRemovesCachedProfile(c *C) {
+func (s *appArmorSuite) TestUnloadRemovesCachedProfileInForest(c *C) {
 	cmd := testutil.MockCommand(c, "apparmor_parser", "")
 	defer cmd.Restore()
 
 	dirs.SetRootDir(c.MkDir())
 	defer dirs.SetRootDir("")
 	err := os.MkdirAll(dirs.AppArmorCacheDir, 0755)
 	c.Assert(err, IsNil)
+	// mock the forest subdir and features file
+	subdir := filepath.Join(dirs.AppArmorCacheDir, "deadbeef.0")
+	err = os.MkdirAll(subdir, 0700)
+	c.Assert(err, IsNil)
+	features := filepath.Join(subdir, ".features")
+	ioutil.WriteFile(features, []byte("blob"), 0644)
 
-	fname := filepath.Join(dirs.AppArmorCacheDir, "profile")
+	fname := filepath.Join(subdir, "profile")
 	ioutil.WriteFile(fname, []byte("blob"), 0600)
 	err = apparmor.UnloadProfiles([]string{"profile"}, dirs.AppArmorCacheDir)
 	c.Assert(err, IsNil)
```

Comment:
We *could* test here that the `.features` file is still there (i.e. that UnloadProfiles() leaves files alone that its not supposed to touch and maybe that the dir itself is still there. But given that its really just a tiny improvement doing that later in a followup is fine.

Reply:
I'll do this now.